### DISCUSSION TITLE
Offline installation of gauge in silent mode

### DISCUSSION THREAD

**Discussion by ayankumaar** [getgauge/gauge#1941] [General]
I have downloaded the gauge windows installer. I am running the following command in the windows command prompt

**gauge-1.1.7-windows.x86_64.exe /S**

This is doing a complete silent installation of gauge. However, I am running after this **gauge install dotnet**. It is not installing the dotnet plugin. I know the reason i.e. my internet is turned off. 

**I need to give a completely silent installation of gauge while my internet is turned off. At the same time I also need to install the dotnet plugin and the html reports plugin.** 

How can I achieve this ?

**Comment by sriv** [comment]
You can download the plugin archives and use offline install - https://docs.gauge.org/plugin.html#install-plugins-offline

**Reply by ayankumaar** [reply]
@sriv thank you very much for your reply. So, you mean I run like this:

First I turn off the internet and then

1) Run    **gauge-1.1.7-windows.x86_64.exe /S**      - This will install gauge in silent mode.

2) Then I should run **gauge install dotnet --file "C://gauge//dotnet_x.x.x.zip"**   - This will install dotnet plugin

Am I right ?

**Reply by sriv** [reply]
yes, that should work

**Reply by ayankumaar** [reply]
@sriv I had another question. I am noticing that if my internet connection is turned on then while doing

**gauge-1.1.7-windows.x86_64.exe /S**    - It is automatically installing the html plugin. How to turn that off while the internet is turned on ? I want the user to install only from the zip file that I give him for HTML plugin.

**Reply by sriv** [reply]
I think if you just need gauge exe, then it's probably better for you to simply use the zip archive, extract the exe, making sure that the location is in PATH.

The installer considers html report a mandatory plugin and will install latest version by default, unless there is no internet connection.

**Reply by ayankumaar** [reply]
@sriv how to modify the default installation folder path while doing silent installation ? I don't want to install it inside Program Files.

Basically, I want to do something like this:

**gauge-1.1.7-windows.x86_64.exe /S /D "Custom_Installation_Folder_Path"**

Is it possible to install that way ?

**Reply by sriv** [reply]
You should really try using the zip archive as I mentioned above. If you don't need to install in program files, and don't want to install plugins, there is no reason to choose the exe installer IMO

**Reply by ayankumaar** [reply]
@sriv Is there any way to install the zip archive from the windows command prompt ? or Powershell is mandatory to install it ?

**Reply by sriv** [reply]
It's a regular zip file. You can use any tool that suits you. Powershell has built in support. With command prompt you may need to use 7zip or similar tools
